index.bs

```diff
@@ -0,0 +1,331 @@
+  Autoplay detection can be performed on either the {{Document}} element or on
+  the {{HTMLMediaElement}}. The former provides a general idea to authors to
+  know if autoplay media is allowed on this document element. If the page
+  contains multiple documents, an implementation can decide to return different
+  result for each of them. The latter provides more accurate result for whether
+  the autoplay is allowed on a specific media element.
+
+  <h3 id="autoplay-policy">Autoplay Policy Enum</h3>
+    <pre class="idl">
+    enum AutoplayPolicy {
+      "allowed",
+      "allowed-muted",
+      "disallowed"
+    };
+    </pre>
+    <div class="enum-description">
+    <table class="simple" dfn-for="AutoplayPolicy" dfn-type="enum-value">
+      <thead>
+        <tr>
+          <th scope="col" colspan="2">
+            Enumeration description
+      <tbody>
+        <tr>
+          <td>
+            "<dfn>allowed</dfn>"
+          <td>
+            Media are allowed to autoplay.
+        <tr>
+          <td>
+            "<dfn>allowed-muted</dfn>"
+          <td>
+            Inaudible media element</dfn> are allowed to autoplay. An
+            <dfn export>inaudible media element</dfn> is an {{HTMLMediaElement}}
+            that has any of those conditions true:
+            <ol>
+            <li> media's {{HTMLMediaElement/volume}} equal to 0
+            <li> media's {{HTMLMediaElement/muted}} is true
+            <li> media's <a href="https://html.spec.whatwg.org/multipage/media.html#media-resource">resource</a> does not have an audio track
+            </ol>
+        <tr>
+          <td>
+            "<dfn>disallowed</dfn>"
+          <td>
+            No media is allowed to autoplay.
+    </table>
+    </div>
+
+    <div class="note">
+      Note: The autoplay policy represents the current status of whether a
+      user-agent allows media to autoplay, which can **vary** in the future.
+      Therefore, it is **recommended** authors to check the result every time
+      they want to have an up-to-date result.
+    </div>
+
+    <div class=example>
+      If a user-agent uses the user activation, described in
+      [[HTML#user-activation-data-model]], to determine if the autoplay media
+      are allowed or not, and it blocks all autoplay by default
+      ({{AutoplayPolicy/disallowed}}). Then the policy can change to
+      {{AutoplayPolicy/allowed}} or {{AutoplayPolicy/allowed-muted}} after a
+      user performs a supported user gesture on the page or the media.
+    </div>
+
+  <h3 id="document-api">The Document API</h3>
+    <pre class="idl">
+    partial interface Document {
+      readonly attribute AutoplayPolicy autoplayPolicy;
+    };
+    </pre>
+    This represents a rough status of whether media that belong to this
+    {{Document}} are allowed to autoplay or not.
+
+    <dl class="switch">
+      <dt>If the value is {{allowed}}</dt>
+      <dd>
+        All media that belong to this document are allowed to autoplay.
```

Comment:
Do you want to say more about has a user gesture might affect this?

Reply:
Does the note or example below meet this? Lines 156 and 166.

Reply:
Yes, I used note and example below to demonstrate the possible situaion of using user gesture activation to detect blocking autoplay for user-agents. @dalecurtis let me know if you think that's not enough or I should add more details in the note and example. In addition, because user-agents could use different strategy for blocking autoplay, eg. Safari doesn't use the user activation model. That makes me figure that avoiding to mention the user gesture in the normative form might be better?

Reply:
In addition, I just think of one potential issue. If the user-agent is using a non-user-activation way for blocking autoplay, then it seems possible that we would have different result for media element and web audio in term of allowing to play?

Eg. 
If a user-agent implements a click-to-play strategy (only allow an media element to start after user clicks on it), so event if a user clicks on the page (not element), the document's `autoplaypolicy` would still be `disallowed`. But web audio uses sticky activation, it should be allowed to start after the page granted the sticky activation. In this situation, if `document.autoplayPolicy` returns `disallowed` then it doesn't reflect the web audio's state. But if it returns `allowed` then it violates the media element's state.

I wonder how y'all think about this case? Thank you.

Reply:
To me it sounds like we shouldn't try and represent both web audio and the media element with a single property.

Maybe some static ones on the actual apis?

Like `HTMLMediaElement.autoplayPolicy` and `AudioContext.autoplayPolicy`?

Reply:
That sounds reasonable to me, let me post a new comment on the thread and ask other people's opinion.